### ISSUE TITLE
bug(larql-cli): 16 NaN-unsafe float sort panics across extraction commands

### ISSUE THREAD

**Issue by metavacua** [metavacua/larql-to-sparql#222]
## Description

16 locations in `crates/larql-cli/src/` use `partial_cmp().unwrap()` for sorting floating-point values. `partial_cmp` returns `None` when either operand is `NaN`, causing `unwrap()` to panic at runtime. This affects any extraction command that receives NaN in model weights or computed scores.

## Affected files

```
crates/larql-cli/src/commands/diagnostics/parity.rs:1126
crates/larql-cli/src/commands/extraction/circuit_discover_cmd.rs:131,133,385
crates/larql-cli/src/commands/extraction/ffn_bottleneck_cmd.rs:106
crates/larql-cli/src/commands/extraction/ffn_overlap_cmd.rs:71
crates/larql-cli/src/commands/extraction/kg_bench_cmd.rs:118
crates/larql-cli/src/commands/extraction/ov_gate_cmd.rs:166,168,366,368
crates/larql-cli/src/commands/extraction/qk_modes_cmd.rs:157,159
crates/larql-cli/src/commands/extraction/qk_templates_cmd.rs:201,258
crates/larql-cli/src/commands/extraction/trajectory_trace_cmd.rs:207
```

## Example

```rust
// Before (panics on NaN):
idx.sort_by(|&a, &b| logits[b].partial_cmp(&logits[a]).unwrap());

// After (NaN sorts to end, consistent behavior):
idx.sort_by(|&a, &b| logits[b].total_cmp(&logits[a]));
```

## Fix

Replace `partial_cmp(...).unwrap()` with `f32::total_cmp` (stable since Rust 1.62, already above the workspace MSRV). `total_cmp` defines a total order including NaN values — NaN sorts consistently to either end.

## Labels

- kind: bug
- area: larql-cli
- priority: medium (extraction tools only; not in inference hot path)
